core/templates/components/common-layout-directives/navigation-bars/side-navigation-bar.component.ts

```diff
@@ -71,34 +70,27 @@ export class SideNavigationBarComponent {
   ngOnInit(): void {
     this.currentUrl = this.windowRef.nativeWindow.location.pathname;
 
-    let service = this.classroomBackendApiService;
-    service.fetchClassroomPromosAreEnabledStatusAsync().then(
-      classroomPromosAreEnabled => {
-        this.CLASSROOM_PROMOS_ARE_ENABLED = classroomPromosAreEnabled;
-        if (classroomPromosAreEnabled) {
-          this.accessValidationBackendApiService.validateAccessToClassroomPage(
-            this.DEFAULT_CLASSROOM_URL_FRAGMENT).then(()=>{
-            this.classroomBackendApiService.fetchClassroomDataAsync(
-              this.DEFAULT_CLASSROOM_URL_FRAGMENT)
-              .then((classroomData) => {
-                this.classroomData = classroomData.getTopicSummaries();
-                this.classroomBackendApiService.onInitializeTranslation.emit();
-                // Store hacky tranlation keys of topics.
-                for (let i = 0; i < this.classroomData.length; i++) {
-                  let topicSummary = this.classroomData[i];
-                  let hackyTopicTranslationKey = (
-                    this.i18nLanguageCodeService.getTopicTranslationKey(
-                      topicSummary.getId(), TranslationKeyType.TITLE
-                    )
-                  );
-                  this.topicTitlesTranslationKeys.push(
-                    hackyTopicTranslationKey
-                  );
-                }
-              });
-          });
-        }
-      });
+    this.accessValidationBackendApiService.validateAccessToClassroomPage(
+      this.DEFAULT_CLASSROOM_URL_FRAGMENT).then(()=>{
```

Comment:
Done

Reply:
Hey @Rijuta-s, after implementing the changes regarding the "warning-raise" I'm getting it raised on the signup page and it is because the `side-navigation-bar component` is part of `oppia-base-content component` and the signup page uses the base component, Please note that the warning is only getting raised on the signup page and nowhere else.
Screenshot of the signup page:
![Screenshot from 2023-06-09 12-12-58](https://github.com/oppia/oppia/assets/44300735/ecc2dcb0-3645-4554-913b-2bbb4c8ef708)

How do I handle this part?
Thanks

Reply:
One thing that I found out after testing in develop with following steps:
1. Login with the admin account and enable the `classroom promos` config property.
2. Logout and try to signup with a new account and you will not be able to, please have a look at the screenshot below
It is stuck at the signup page
![Screenshot from 2023-06-10 13-20-59](https://github.com/oppia/oppia/assets/44300735/3d1565c7-30e9-4c11-b0f4-7b5ef8b5982c)

Now it makes sense why the error mentioned in my previous comment is appearing because I'm turning on the "classroom promos" permanently.

Reply:
We will need to use an if condition or something like that to check if the side bar actually needs to be loaded or not, why is the data for the sidebar getting loaded even when it is not required ?

Reply:
Actually both the `side-bar` and the `top-bar` are part of the `base-content` component and the base component is getting loaded everywhere hence we are automatically loading the data of both the bar component.
I think it will be a bit hard to find a way for when to load and when not to load, afaik there should be some way to tell the `base-content` component html that it is getting loaded for the signup page and we should then avoid the loading the bar components, not sure?
Thanks

Reply:
Why is there an issue now when there wasn't before? Classroom promos have been enabled for ages in production with no error.

Reply:
Hmm, but do we have a signup page on production? I mean when we try to login we are only allowed to login via the google accounts and that is why the signup page does not appear hence we are not aware of the errors if they appear.
Is there a way to signup via email and username like we do in dev mode and I'm not aware of it? By looking at the code I think we can only sign in via the google accounts.

Thanks

Reply:
We do have a signup page on production. It appears after you sign in via the google account; you are then redirected to this page where you need to choose a username and agree to the T+C.

Reply:
Ah, okay got it.
So, I checked and we do have similar problem on the production:
![Screenshot from 2023-06-11 10-42-39](https://github.com/oppia/oppia/assets/44300735/6382144b-3247-4376-a820-31ff9da0caae)

Thanks

Reply:
Yes that's fine. Thanks.

Reply:
Yeah, seems fine to me too, adding a TODO and opening an issue for it, works.